src/sentry/static/sentry/app/components/events/relatedEvents/index.tsx

```diff
@@ -0,0 +1,278 @@
+
+    return EventView.fromSavedQuery({
+      id: undefined,
+      name: `Events with Trace ID ${traceId}`,
+      fields: [
+        'title',
+        'event.type',
+        'project',
+        'project.id',
+        'trace.span',
+        'timestamp',
+        'lastSeen',
+        'issue',
+      ],
+      orderby: '-timestamp',
+      query: `trace:${traceId}`,
+      projects: orgFeatures.has('global-views') ? [-1] : [Number(event.projectID)],
+      version: 2,
+      range: '90d',
+    });
+  };
+
+  const getTransactionLink = (projectId: string, transationName: string) => {
+    return transactionSummaryRouteWithQuery({
+      orgSlug,
+      transaction: transationName,
+      projectID: projectId,
+      query: {
+        statsPeriod: '90d',
+      },
+    });
+  };
+
+  const getEventTarget = (
+    dataRow: TableDataRow,
+    eventView: EventView,
+    currentLocation: CURRENT_LOCATION
+  ) => {
+    if (currentLocation === CURRENT_LOCATION.DISCOVER) {
+      const eventSlug = generateEventSlug(dataRow);
+
+      return eventDetailsRouteWithEventView({
+        orgSlug,
+        eventSlug,
+        eventView,
+      });
+    }
+
+    return dataRow['event.type'] === 'error'
+      ? `/organizations/${orgSlug}/issues/${dataRow['issue.id']}/`
+      : getTransactionLink(String(dataRow['project.id']), String(dataRow.title));
+  };
+
+  const renderEventId = (
+    dataRow: TableDataRow,
+    eventView: EventView,
+    currentLocation: CURRENT_LOCATION
+  ) => {
+    return (
+      <Tooltip title={t('View Event')}>
+        <StyledLink to={getEventTarget(dataRow, eventView, currentLocation)}>
+          {dataRow.id as React.ReactNode}
+        </StyledLink>
+      </Tooltip>
+    );
+  };
+
+  const renderActions = (eventView: EventView, currentLocation: CURRENT_LOCATION) => {
+    if (
+      !orgFeatures.has('discover-basic') ||
+      currentLocation === CURRENT_LOCATION.DISCOVER
+    ) {
+      return undefined;
+    }
+
+    const discoverURL = eventView.getResultsViewUrlTarget(orgSlug);
+
+    return (
+      <Button size="small" to={discoverURL}>
+        {t('Open in Discover')}
+      </Button>
+    );
+  };
+
+  const renderEmptyMessage = () => {
+    const {
+      datetime: {period},
+    } = selection;
+
+    const selectedTimePeriod = period && DEFAULT_RELATIVE_PERIODS[period];
+    const displayedPeriod = selectedTimePeriod
+      ? selectedTimePeriod.toLowerCase()
+      : t('given timeframe');
+
+    return tct('No related events have been found for the [timePeriod].', {
+      timePeriod: displayedPeriod,
+    });
+  };
+
+  const renderIcon = (type: EVENT_TYPE) => {
+    if (type === EVENT_TYPE.ERROR) {
+      return <IconFire color="red400" />;
+    }
+    return <IconSpan color="pink400" />;
+  };
+
+  const eventView = getEventView();
+  const currentLocation = getCurrentLocation();
+
+  return (
+    <DiscoverQuery location={location} eventView={eventView} orgSlug={orgSlug}>
```

Comment:
This request could fail when accounts don't have access to the discover endpoints. Should there be a path for those folks where this component renders `null`?